protocol/lavasession/single_provider_session.go

```diff
@@ -107,7 +107,7 @@ func (sps *SingleProviderSession) PrepareDataReliabilitySessionForUsage(relayReq
 	return nil
 }
 
-func (sps *SingleProviderSession) PrepareSessionForUsage(ctx context.Context, cuFromSpec uint64, relayRequestTotalCU uint64, allowedThreshold float64) error {
+func (sps *SingleProviderSession) PrepareSessionForUsage(ctx context.Context, cuFromSpec uint64, relayRequestTotalCU uint64, allowedThreshold float64, badgeUser string) error {
```

Comment:
lets pass the *badgeSession object here as well (instead of the badge user string)

Reply:
Fixed here: 704563ba77bb3d13a013adcc88dce17aba97627a